refactor(connector): remove default case handling

HeetVekariya opened the pull request:
## Type of Change
<!-- Put an `x` in the boxes that apply -->

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [X] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
<!-- Describe your changes in detail -->
- Fix issue #2278 
- Changes to be made : Remove default case handling
- Added all the ```Banknames``` (which are not ```NuveiBIC``` ) against the default case
- File changed: **```transformers.rs```** (crates/router/src/connector/nuvei/transformers.rs)

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

<!--
Provide links to the files with corresponding changes.

Following are the paths where you can find config files:
1. `config`
2. `crates/router/src/configs`
3. `loadtest/config`
-->


## Motivation and Context
<!--
Why is this change required? What problem does it solve?
If it fixes an open issue, please link to the issue here.

If you don't have an issue, we'd recommend starting with one first so the PR
can focus on the implementation (unless it is an obvious bug or documentation fix
that will have little conversation).
-->


## How did you test it?
<!--
Did you write an integration/unit/API test to verify the code changes?
Or did you test this change manually (provide relevant screenshots)?
-->


## Checklist
<!-- Put an `x` in the boxes that apply -->

- [ ] I formatted the code `cargo +nightly fmt --all`
- [ ] I addressed lints thrown by `cargo clippy`
- [ ] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

SanchithHegde commented:
@HeetVekariya Can you address the failing formatting check? You can install the nightly toolchain and run `cargo +nightly fmt` to format code.

HeetVekariya commented:
- @SanchithHegde Changes made, passed all the checks.

prasunna09 commented on crates/router/src/connector/nuvei/transformers.rs:
```diff
@@ -490,11 +490,145 @@ impl TryFrom<api_models::enums::BankNames> for NuveiBIC {
+            | api_models::enums::BankNames::TsbBank
+            | api_models::enums::BankNames::TescoBank
+            | api_models::enums::BankNames::UlsterBank => {
+                Err(errors::ConnectorError::FlowNotSupported {
+                    flow: bank.to_string(),
+                    connector: "Nuvei".to_string(),
+                }
+                .into())
```
```suggestion
                {
                 Err(errors::ConnectorError::NotSupported {
                     message: bank.to_string(),
                     connector: "Nuvei".to_string(),
                 }
                 .into())
```

HeetVekariya commented:
- @prasunna09 I have implemented the suggested changes.
- While implementing the changes i found some error:
![Screenshot from 2023-10-16 07-22-09](https://github.com/juspay/hyperswitch/assets/91054457/0d1bba11-b798-412c-a1aa-5b0ccbde0f80)

- The error says that, ```connector``` prop is ```static str``` which does not allow to use ```to_string()``` method
- So i have removed it

SanchithHegde reviewed (approved):
Looks good to me!

Thanks for the PR, @HeetVekariya!

swangi-kumari commented:
Hey @HeetVekariya,
Thanks a bunch for all your contributions! We've got a little something for you to show our appreciation.  
Just take a moment to fill out this [form](https://forms.gle/yqbtjFuP6oxLaBKN8), and get ready for some awesome swag coming your way.  
Thanks!

HeetVekariya commented:
- @swangi-kumari  I have a great time contributing to this repo.
- Thank you for the swag
- I will continue to contribute :+1:
